### DISCUSSION TITLE
提供 macOS 版本的執行檔案 (同步到 7.2 3/29 v3.3.2.1)

### DISCUSSION THREAD

**Discussion by hydai** [GpointChen/FFXIVChnTextPatch-GP#342] [Show and tell]
我自行編譯 macOS 的版本，希望可以幫助其他使用 macOS 的玩家
目前是使用 [GitHub action](https://github.com/hydai/FFXIVChnTextPatch-GP/blob/hydai/.github/workflows/build-on-macos.yml) 進行編譯與發佈，如有任何建議都歡迎討論，謝謝

https://github.com/hydai/FFXIVChnTextPatch-GP/releases/latest

---
## 安裝方式

請根據「你的硬體」與「想使用的字體版本」下載對應的 dmg 檔案

### 普通字體

```
FFXIVChnTextPatch-GP-Intel.dmg # <--- 如果你是 Intel 版的 mac
FFXIVChnTextPatch-GP-AppleSilicon.dmg # <--- 如果你是 AppleSilicon(M1/M2/Mx) 版的 mac
```

### 4K 字體

```
FFXIVChnTextPatch-GP-4k-Intel.dmg # <--- 如果你是 Intel 版的 mac
FFXIVChnTextPatch-GP-4k-AppleSilicon.dmg # <--- 如果你是 AppleSilicon(M1/M2/Mx) 版的 mac
```

## 使用方式

1. 點兩下下載的檔案，掛載 dmg 以後
2. 請將 `FFXIVChnTextPatchGP.app` 拖進 `Applications` 資料夾
3. 請到 `Applications` 資料夾中找到 `FFXIVChnTextPatchGP.app` 並點兩下執行
4. 如果是第一次執行，應該會遇到安全性問題，請參考下方的注意事項
5. 此時如果顯示「無法打開應用程式「FFXIVChnTextPatch-GP」。」，請參考下方的注意事項

#### 安全性問題

如遇到無法執行的問題，請到系統偏好設定 -> 安全性與隱私 -> 通用，將「允許從以下位置下載的 App」設定為「任何來源」，再重新執行 App 即可。

或參考蘋果官方文章來操作： https://support.apple.com/zh-tw/guide/mac-help/mchleab3a043/mac

#### 無法打開應用程式

請打開終端機，並執行以下指令：

```bash
# 請確保執行此指令前，已經做完 1. 安全性問題設定 2. 將 App 拖進 Applications 資料夾

/Applications/FFXIVChnTextPatch-GP.app/Contents/MacOS/universalJavaApplicationStub
```

**Comment by hydai** [comment]
對應 6.4 5/25 v3.1.0.1 release 的版本：

https://github.com/hydai/FFXIVChnTextPatch-GP/releases/tag/v0.0.2

**Comment by hydai** [comment]
對應 6.4 5/26 v3.1.0.2 release 的版本：

https://github.com/hydai/FFXIVChnTextPatch-GP/releases/tag/v0.0.3

**Comment by mslauky** [comment]
感謝樓主分享
可是我卡在font6.tex 請問有解決方法嗎？
iMac (Late 2015, macOS Monterey)

<img width="288" alt="220130025-2e89f7a6-b497-4f3a-b592-4878c3f9f3f6" src="https://github.com/GpointChen/FFXIVChnTextPatch-GP/assets/62352923/d510b0f5-7477-4d00-91b2-a05d3fbc9038">

```
Exception in thread "Thread-1" java.lang.NoSuchMethodError: java.nio.ByteBuffer.position(I)Ljava/nio/ByteBuffer;
	at name.yumao.ffxiv.chn.builder.TexBlockBuilder.buildBlock(TexBlockBuilder.java:38)
	at name.yumao.ffxiv.chn.replace.ReplaceFont.replace(ReplaceFont.java:70)
	at name.yumao.ffxiv.chn.thread.ReplaceThread.run(ReplaceThread.java:46)
	at java.lang.Thread.run(Thread.java:750)
```

**Reply by hydai** [reply]
你用的是 FF14 官方的 launcher 嗎？
我看之前那一串也有一樣的問題： https://github.com/GpointChen/FFXIVChnTextPatch-GP/discussions/202
好像都是使用官方的 launcher 的案例

**Reply by GpointChen** [reply]
應該是compile和執行時的java版本不同的問題

可以參考：
* https://github.com/skylot/jadx/issues/1793
* https://stackoverflow.com/questions/61267495/exception-in-thread-main-java-lang-nosuchmethoderror-java-nio-bytebuffer-flip
* https://www.morling.dev/blog/bytebuffer-and-the-dreaded-nosuchmethoderror/

**Reply by hydai** [reply]
原來是跟 java 版本有關，CI 上面用 openjdk@11 進行編譯的，請問 @mslauky 你安裝的 java 版本是什麼呢？

另外，我嘗試了用 `javapackager` 把 JRE 一起包進去，看能不能解決你的問題

下載這個，裡面有包裝好的 dmg 檔案
![CleanShot 2023-06-01 at 03 49 40](https://github.com/GpointChen/FFXIVChnTextPatch-GP/assets/2776756/e2033ff6-e8e3-44a2-9b65-75aa7d9d13b8)

https://github.com/hydai/FFXIVChnTextPatch-GP/actions/runs/5136903812

**Reply by mslauky** [reply]
感謝樓主和作者幫忙

以下是我的java版本
<img width="376" alt="螢幕截圖 2023-06-01 下午5 42 49" src="https://github.com/GpointChen/FFXIVChnTextPatch-GP/assets/62352923/c530fa69-7ca7-4fad-8eac-4edfa04fffb6">

對的 我是使用官方launcher然後就卡了...
最後我是找弄了漢化的windows玩家打包6個000檔案給我 然後手動放進xivonmac launcher文件夾解決了問題
（官方launcher實在是太爛 氣死了... 轉用xivonmac穩定很多）
我晚點再試一下hydai的dmg檔 萬分感激！

 另外想問一下@GpointChen以後能不能直接分享那6個000檔案在github？ 
因為之前有windows朋友使用exe之後電腦直接當掉 遊戲又要重新安裝（可能他是個別案件吧）
手動漢化也好像比較方便mac玩家使用

**Reply by hydai** [reply]
這個是 Java 8 ，我用的是 OpenJDK 11，看起來真的是使用的版本差異，可能還是要跟作者在 Windows 發行的那個執行檔案一樣想辦法打包 JRE 進去會更合適

我個人是一開始就使用 xivonmac ，主要是發現他執行起來的效能比官方的好太多了

**Reply by mslauky** [reply]
剛剛再弄一次`brew install openjdk@11`發現jdk都已經安裝好了

```
openjdk@11 11.0.19 is already installed and up-to-date.
To reinstall 11.0.19, run:
  brew reinstall openjdk@11
```

**Reply by hydai** [reply]
你系統裡面應該同時有 1. Java 8; 2. OpenJDK 11. 然後執行時應該是優先選了 Java 8，如果上面的 dmg 也失敗的話，那還需要另外設置環境做實驗了

**Reply by mslauky** [reply]
@hydai 剛剛用了你做的dmg終於能跑起來！（遊戲路徑是官方launcher）沒再卡在font6了！超級感動！
進遊戲測試運作正常 沒有dx大爆炸
那麼想問一下 下次弄漢化也可以用這個dmg還是要麻煩你弄一個新的？

**Reply by hydai** [reply]
我之後也會釋出這個 dmg 隨著作者的版本一起更新，謝謝你幫忙測試！

**Reply by mslauky** [reply]
太好了！非常感謝你弄的dmg檔案～ 
也想問一下xivonmac的路徑是在哪裡，不想每次在官方launcher這邊弄好漢化後再手動放進去

**Reply by hydai** [reply]
@mslauky 
XIVonMac 的路徑是這個 `/Users/使用者帳號/Library/Application Support/XIV on Mac/ffxiv`

**Reply by shiny10cean** [reply]
大大您好～我也是遇到這個問題卡在這的人，有看了一下新上傳的解決方法dmg 載點但過期了
請問能再跟大大要一次嗎？萬分感謝～
<img width="289" alt="截圖 2023-09-13 下午6 57 47" src="https://github.com/GpointChen/FFXIVChnTextPatch-GP/assets/144900232/1cb32cd8-802d-404e-8541-f93f5dba259b">

**Reply by hydai** [reply]
最新版本已經發布，請下載對應的版本來使用

**Reply by shiny10cean** [reply]
大大~還是卡在一樣的地方T_T

**Reply by hydai** [reply]
你的環境是什麼？
具體的操作步驟是？

**Reply by shiny10cean** [reply]
想問一下還是是不是因為我的是體驗版的關係？＠＠

**Reply by hydai** [reply]
跟體驗版無關，不管有沒有月卡都可以進行中文化

**Reply by shiny10cean** [reply]
<img width="275" alt="截圖 2023-09-17 下午10 48 10" src="https://github.com/GpointChen/FFXIVChnTextPatch-GP/assets/144900232/2647a547-4d68-4a80-b638-e49557ff1428">
<img width="375" alt="截圖 2023-09-13 下午10 30 57" src="https://github.com/GpointChen/FFXIVChnTextPatch-GP/assets/144900232/5c143ef2-1b98-4e07-b772-b7102b87ea46">

**Reply by shiny10cean** [reply]
我是先下載 FFXIVChnTextPatch-GP-AppleSilicon.dmg
然後下載 [FFXIVChnTextPatch-GP-macOS.zip]

先 FFXIVChnTextPatch-GP-AppleSilicon.dmg 打開 丟進app 然後去安全性強制打開 不過載完貌似開不起來
在開 FFXIVChnTextPatch-GP-macOS.zip 裡面那執行檔 選漢化 選資料夾是什麼reborn的 然後英文 就卡在 font6

**Reply by hydai** [reply]
你只需要 `FFXIVChnTextPatch-GP-AppleSilicon.dmg` 就好了，不需要額外下載 `macOS` 的那個 zip

**Reply by shiny10cean** [reply]
那好像打不開＠＠ 一開起來又自己消失

**Reply by shiny10cean** [reply]
啊 剛剛我把終端機關掉之後 他變成只會出現在 dock 上 但什麼也沒跑出來

**Reply by hydai** [reply]
請參考： https://support.apple.com/zh-tw/guide/mac-help/mh40616/mac

**Reply by shiny10cean** [reply]
有了已經強制打開 目前就是會出現在 dock 上但什麼都沒跑出來

**Reply by shiny10cean** [reply]
<img width="89" alt="截圖 2023-09-17 下午11 48 47" src="https://github.com/GpointChen/FFXIVChnTextPatch-GP/assets/144900232/ca7fe53b-1c72-4984-a796-b87c5372cac9">

**Reply by hydai** [reply]
看起來沒有打開嗎？
我的在強制打開後會出現對應的對話框
![CleanShot 2023-09-17 at 23 50 04](https://github.com/GpointChen/FFXIVChnTextPatch-GP/assets/2776756/438215e9-5a62-401f-967d-a83f99480c21)

**Reply by shiny10cean** [reply]
我的沒有嗚嗚嗚

**Reply by hydai** [reply]
> 啊 剛剛我把終端機關掉之後 他變成只會出現在 dock 上 但什麼也沒跑出來

先把不相關的部分都關掉，把 dmg 掛載以後，點擊 `app` 兩下應該會是跳出安全性的通知，然後去設定那邊強制打開，後續就可以啟動了，我剛才拿了另一台沒用過的 M1 macbook 來測試也是可以正常執行

**Reply by shiny10cean** [reply]
重開機全關了還是一樣 哭

**Reply by shiny10cean** [reply]
最後按了強制打開跑出來是這樣

<img width="257" alt="截圖 2023-09-18 上午12 47 48" src="https://github.com/GpointChen/FFXIVChnTextPatch-GP/assets/144900232/a4103b3a-8a47-406b-b263-9a0a3604993f">

**Reply by hydai** [reply]
強制打開後，點選打開，沒有任何反應嗎？

**Reply by shiny10cean** [reply]
對的～～～

**Reply by seventhmoon27** [reply]
我也是同樣問題。打不開

**Reply by hydai** [reply]
@seventhmoon27 
你是什麼平台的？使用哪個版本的 dmg？

**Reply by seventhmoon27** [reply]
MacOS: Sonoma 14.2.1
M2 Macbook air
用了silicon的.dmg

FFXIVChnTextPatch-GP 在dock上，Patch Tools不出來

**Reply by hydai** [reply]
@seventhmoon27 
那個 dmg 是在 M2 Max macbook pro + Sonoma 14.2.1 來製作的
你先把他拉到 application 裡面再執行試試看？
他會一直在 dock 上跳動還是就只有出現圖示，然後沒有 GUI 跳出來呢？

**Reply by seventhmoon27** [reply]
有拉到application，然後强制打開。icon在dock上跳動兩下，沒有GUI。

**Reply by hydai** [reply]
我沒辦法重現這個問題，目前手上有四台 macbook:
1. M1 Max macbook pro
2. M2 Max macbook pro
3. Intel i9 macbook pro 2019
4. Intel i5 macbook pro 2015

剛才抓下來測試都可以正常執行，可能要研究一下怎麼樣導出錯誤資訊來判斷問題在哪了

**Comment by hydai** [comment]
對應 6.4 5/26 v3.1.0.2 release 的版本：

https://github.com/hydai/FFXIVChnTextPatch-GP/releases/tag/v0.0.4

目前已經將 dmg (Intel / Apple Silicon) 的版本都更新上去囉

**Comment by asit0221** [comment]
您好，請問試玩版也可以用它來漢化嗎＞＜
因為我不太確定遊戲根目錄要選哪一個，也沒有上面XIVonMac的路徑

我是macbook air m1版本的

**Reply by hydai** [reply]
你用的是官方版本還是 XIVonMac?
以路徑來說：
官方：「`/Users/使用者帳號/Library/Application\ Support/FINAL\ FANTASY\ XIV\
ONLINE/Bottles/published_Final_Fantasy/drive_c/Program\ Files\
\(x86\)/SquareEnix/FINAL\ FANTASY\ XIV\ -\ A\ Realm\ Reborn`」
XIVonMac：「`/Users/使用者帳號/Library/Application Support/XIV on Mac/ffxiv`」

**Reply by asit0221** [reply]
我是用官方版本的！
剛剛才釐清兩者的不同，早上我再試試看能不能裝，謝謝您～

**Comment by hydai** [comment]
已更新對應 6.4 6/8 v3.1.1.0 release 的版本：

https://github.com/hydai/FFXIVChnTextPatch-GP/releases/tag/v0.0.5

**Comment by hydai** [comment]
已更新對應 6.41 6/20 v3.1.1.1 release 的版本：

https://github.com/hydai/FFXIVChnTextPatch-GP/releases/tag/v0.0.6

**Comment by mslauky** [comment]
請問會有對應6.48 8/13的更新嗎？

**Reply by hydai** [reply]
弄好了～

**Comment by hydai** [comment]
已更新對應 6.48 8/13 v3.1.1.5 release 的版本：

https://github.com/hydai/FFXIVChnTextPatch-GP/releases/tag/v0.0.8

**Comment by hydai** [comment]
更新對應 6.48 9/10 v3.1.1.6 release 的版本：

https://github.com/hydai/FFXIVChnTextPatch-GP/releases/tag/v0.0.9

**Comment by mslauky** [comment]
請問會有對應6.5 10/3的更新嗎？感謝大大幫忙～

**Reply by hydai** [reply]
更新好了

**Comment by hydai** [comment]
更新對應 6.51 10/31 v3.2.0.3 release 的版本：

https://github.com/hydai/FFXIVChnTextPatch-GP/releases/tag/v0.1.1

**Comment by hydai** [comment]
從 v0.1.2 開始（對應 6.51 10/31 v3.2.0.3 release）提供 4k 字體版本，請根據 wiki 做出相對應的設定調整後使用

**Reply by mslauky** [reply]
感謝分享

**Comment by hydai** [comment]
更新對應 6.51 11/30 v3.2.0.4 release 的版本：

https://github.com/hydai/FFXIVChnTextPatch-GP/releases/tag/v0.1.3

**Comment by hydai** [comment]
更新對應 6.51 1/1 v3.2.1.1 release 的版本：

https://github.com/hydai/FFXIVChnTextPatch-GP/releases/tag/v0.1.4

**Comment by hydai** [comment]
更新對應 6.58 3/19 v3.2.1.5 release 的版本：

https://github.com/hydai/FFXIVChnTextPatch-GP/releases/tag/v0.1.5

感謝 GitHub，目前已經可以完全使用 CI 把 macos Intel/AppleSilicon 的版本都編譯出來了

**Reply by mslauky** [reply]
謝謝～

**Reply by onezi01** [reply]
![截圖 2024-04-04 下午2 45 17](https://github.com/GpointChen/FFXIVChnTextPatch-GP/assets/166003950/92e8b602-cc96-4d04-a020-0d20aa0ade34)
我還是碰到一樣的問題，爬了一下java已更新

**Reply by hydai** [reply]
疑？目前的封裝已經把編譯時使用的 Java 嵌入 dmg 裡面了，不確定為什麼還是會發生這個問題

**Reply by onezi01** [reply]
恩，我後來是另外一篇用手動的找檔案覆蓋～
還是很感謝辛苦更新．．mac太容易不知道哪個毛出問題？

**Reply by mslauky** [reply]
請問大佬會持續更新mac版本的漢化嗎？
最近7.0更新了 試過手動覆蓋會出現dx11的問題 遊戲跑不了

**Reply by hydai** [reply]
更新了

**Comment by william7928** [comment]
<img width="396" alt="截圖 2024-06-22 晚上8 31 22" src="https://github.com/GpointChen/FFXIVChnTextPatch-GP/assets/127039071/32ae0d53-020f-472f-8dad-305fe746ee42">
遇到這個問題 打不開檔案

**Reply by hydai** [reply]
@william7928 
請參考蘋果官方文件解決此問題： https://support.apple.com/zh-tw/guide/mac-help/mchleab3a043/mac

**Comment by hydai** [comment]
更新到 7.0 7/3 v3.3.0.1: https://github.com/hydai/FFXIVChnTextPatch-GP/releases/tag/v0.2.1
最近在跑 7.0 分身乏術，剛空出時間來更新(=ﾟωﾟ)ﾉ

**Comment by hydai** [comment]
更新到 7.0 7/12 v3.3.0.3: https://github.com/hydai/FFXIVChnTextPatch-GP/releases/tag/v0.2.2

**Comment by hydai** [comment]
更新到 7.01 7/17 v3.3.0.4: https://github.com/hydai/FFXIVChnTextPatch-GP/releases/tag/v0.2.3

**Comment by hydai** [comment]
更新到 7.05 10/05 v3.3.1.1: https://github.com/hydai/FFXIVChnTextPatch-GP/releases/tag/v0.2.4

**Comment by hydai** [comment]
更新到 7.15 12/18 v3.3.1.4: https://github.com/hydai/FFXIVChnTextPatch-GP/releases/tag/v0.2.6

**Reply by noa032** [reply]
您好 點擊程式時會跳出無法運行應用程式，安全性部分也允許程式執行了，請問是否有下載步驟漏了呢？
<img width="345" alt="image" src="https://github.com/user-attachments/assets/a2ffe54a-6ecd-4d63-a823-087a19725513" />

**Reply by hydai** [reply]
Hi @noa032 
看起來 macOS 似乎修改了安全性的設定，我要有空閒時間才能研究要怎麼處理了QQ

**Reply by noa032** [reply]
好的！感謝分享！

**Comment by lnkneo** [comment]
你好，請問目前會考慮更新7.2版本的漢化嗎，感謝！

**Comment by hydai** [comment]
更新 7.2 版本的支援，同時也在說明處提供如果執行時發生問題應該怎麼排除的問答

7.2 3/29 v3.3.2.1 同步

https://github.com/hydai/FFXIVChnTextPatch-GP/releases/tag/v0.3.2
